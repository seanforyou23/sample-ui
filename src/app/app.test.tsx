import * as React from 'react';
import App from '.';
import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

describe('App tests', () => {
  test('should render default App component', () => {
    const view = shallow(<App appName="Sample UI" />);
    expect(view).toMatchSnapshot();
  });

  test('should render the dashboard route', () => {
    const view = shallow(
      <MemoryRouter initialEntries={[{
        pathname: '/',
        key: 'dashboard'
      }]}>
        <App appName="Sample UI" />
      </MemoryRouter>
    );
    expect(view).toMatchSnapshot();
  });
});
