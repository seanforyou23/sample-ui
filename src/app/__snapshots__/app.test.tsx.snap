// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should render default App component 1`] = `
<Fragment>
  <BackgroundImage
    className=""
    src={
      Object {
        "filter": "test-file-stub#image_overlay",
        "lg": "test-file-stub",
        "sm": "test-file-stub",
        "sm2x": "test-file-stub",
        "xs": "test-file-stub",
        "xs2x": "test-file-stub",
      }
    }
  />
  <Component
    header={<Unknown />}
    sidebar={<Sidebar />}
  >
    <Routes />
  </Component>
</Fragment>
`;

exports[`App tests should render the dashboard route 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "dashboard",
          "pathname": "/",
          "search": "",
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "dashboard",
        "pathname": "/",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <App
    appName="Sample UI"
  />
</Router>
`;
